test/sql/resultset.sql

```diff
@@ -981,78 +981,50 @@ SELECT * FROM check_test(
-            fail('whatever'),
-            false,
-            'results_eq(values, values) subtle mismatch',
-            'whatever',
-            ''
-        ) AS a(b) LOOP RETURN NEXT tap.b; END LOOP;
-
-        FOR tap IN SELECT * FROM check_test(
-            fail('whatever'),
-            false,
-            'results_eq(values, values) integer type mismatch',
-            'whatever',
-            ''
-        ) AS a(b) LOOP RETURN NEXT tap.b; END LOOP;
-
-    ELSE
-        -- 8.4 does true record comparisions, yay!
-        FOR tap IN SELECT * FROM check_test(
-            results_eq(
-                'VALUES (1, ''foo''::varchar), (2, ''bar''::varchar)',
-                'VALUES (1, ''foo''), (2, ''bar'')'
-            ),
-            false,
-            'results_eq(values, values) subtle mismatch',
-            '',
-            '    Number of columns or their types differ between the queries' ) AS a(b) LOOP
-            RETURN NEXT tap.b;
-        END LOOP;
+-- Handle failure due to more subtle column mismatch
+SELECT * FROM check_test(
+    results_eq(
+        'VALUES (1, ''foo''::varchar), (2, ''bar''::varchar)',
+        'VALUES (1, ''foo''), (2, ''bar'')'
+    ),
+    false,
+    'results_eq(values, values) subtle mismatch',
+    '',
+    '    Number of columns or their types differ between the queries
+        ERROR: cannot compare dissimilar column types character varying and text at record column 2'
+);
 
-        FOR tap IN SELECT * FROM check_test(
-            results_eq(
-                'VALUES (1::int), (2::int)',
-                'VALUES (1::bigint), (2::bigint)'
-            ),
-            false,
-            'results_eq(values, values) integer type mismatch',
-            '',
-            '    Number of columns or their types differ between the queries' ) AS a(b) LOOP
-            RETURN NEXT tap.b;
-        END LOOP;
-    END IF;
-    RETURN;
-END;
-$$ LANGUAGE plpgsql;
-SELECT * FROM subtlefail();
+SELECT * FROM check_test(
+    results_eq(
+        'VALUES (1::int), (2::int)',
+        'VALUES (1::bigint), (2::bigint)'
+    ),
+    false,
+    'results_eq(values, values) integer type mismatch',
+    '',
+    '    Number of columns or their types differ between the queries
+        ERROR: cannot compare dissimilar column types integer and bigint at record column 1'
+);
 
 -- Handle failure due to column count mismatch.
 SELECT * FROM check_test(
     results_eq(
         'SELECT pk from dubs ORDER BY pk LIMIT 2',
-        'SELECT name, pk from dubs ORDER BY pk LIMIT 2'
+        'SELECT pk, name from dubs ORDER BY pk LIMIT 2'
     ),
     false,
     'results_eq(values, values) fail column count',
     '',
-    CASE WHEN pg_version_num() < 80400 THEN '    Results differ beginning at row 1:' ELSE '    Number of columns or their types differ between the queries:' END || '
+    '    Number of columns or their types differ between the queries:
         have: (1)
-        want: (foo,1)'
+        want: (1,foo)
```

Comment:
The "different column count" tests were all somehow "broken" before, as they were throwing `cannot compare dissimilar column types integer and text at record column 1`. I changed the order of columns in all those tests, so that the type of the first column is the same and only the number difference remains - in this case the expected error below is thrown.